### ISSUE TITLE
Support for Dependencies with Multiple Exports

### ISSUE THREAD

**Issue by phobetron** [nicolo-ribaudo/yarn-plugin-conditions#5]
I've noticed that the `exports` field in the `package.json` of conditional dependencies may be completely ignored. For instance, when switching from Vue2 to Vue3, only the `main` field's script is picked up, but none of Vue3's `exports` are available.

Is this problem related to the "Only CommonJS dependencies are supported at the moment" warning? I believe each of those `exports` are CommonJS, but if the issue is actually that only the `main` field is supported, that would make sense.

**Comment by nicolo-ribaudo** [comment]
Yes, this currently only supports "old-style CJS packages" because it creates a "proxy" dependency that looks like this:
```js
// index.js
module.exports = condition ? require("package-a") : require("package-b");
```

It would probably be possible to support `"exports"` by creating multiple proxy files within the package, one for each `"exports"` entry point.

**Comment by phobetron** [comment]
Would those then be referenced in an `exports` field in the proxy's `package.json` file? 

I'm poking around at it a bit, but I don't have an previous experience with Yarn plugins, and I had only been using classic Yarn until about a month ago 😅

**Comment by nicolo-ribaudo** [comment]
Here is how the proxy package is defined: https://github.com/nicolo-ribaudo/yarn-plugin-conditions/blob/daf427b206efb1d56ce93ed28983496cef6ee012/sources/zipUtils.ts#L30-L39

you can probably modify that function to accept a map filename->contents.

Btw, this repo doesn't have any test. If you open a PR, I'll just make sure that it doesn't break the existing projects where I'm using this plugin :sweat_smile:
